### DISCUSSION TITLE
Git fetch and merge ONE file only

### DISCUSSION THREAD

**Discussion by radovan-urban** [community/community#23783] [New to GitHub]
<p>Greetings!</p>
<p>Just getting started with git and GitHub.  I am mostly working on my local copy (only master branch for now), pushing to origin occasionally.  But today I had bunch of local commits (not pushed) and I updated README.md on the GitHub web site resulting in two different versions.</p>
<p>I figured out</p>
<pre><code class="lang-auto">git fetch
</code></pre>
<p>and</p>
<pre><code class="lang-auto">git diff master origin/master
</code></pre>
<p>I could see new files added and README with all deletions … I was hoping to do something like</p>
<pre><code class="lang-auto">git merge origin/master/README.md
</code></pre>
<p>Is this possible?</p>
<p>On a related note:<br>
What is the difference between “git diff master origin/master” and “git diff origin/master master”?  They looked orthogonal to each other … deletions became additions and vice versa.</p>
<p>At any rate I ended up coping REDAME additions from web, doing</p>
<pre><code class="lang-auto">git push --force
</code></pre>
<p>Editing on-line</p>
<pre><code class="lang-auto">git fetch &amp;&amp; git merge origin/master
</code></pre>
<p>It worked but it was awkward …</p>
<p>Thanks.<br>
Radovan</p>

**Comment by airtower-luna** [comment]
<aside class="quote no-group" data-username="radovan-urban" data-post="1" data-topic="117150">
<div class="title">
<div class="quote-controls"></div>
 radovan-urban:</div>
<blockquote>
<p>I was hoping to do something like</p>
<pre><code class="lang-auto">git merge origin/master/README.md
</code></pre>
<p>Is this possible?</p>
</blockquote>
</aside>
<p>I’m not sure what you mean by that, could you explain? From your description up to that point I would’ve assumed that <code>git merge origin/master</code> would have been the way to go to include the change made via web in your local branch.</p>
<aside class="quote no-group" data-username="radovan-urban" data-post="1" data-topic="117150">
<div class="title">
<div class="quote-controls"></div>
 radovan-urban:</div>
<blockquote>
<p>What is the difference between “git diff master origin/master” and “git diff origin/master master”? They looked orthogonal to each other … deletions became additions and vice versa.</p>
</blockquote>
</aside>
<p>That’s exactly what they are. When called on two trees (in this case branches) <code>git diff</code> compares the files in each tree. What is “added” or “removed” depends on which way around you look at them.</p>

**Comment by radovan-urban** [comment]
<p>Thank you for a quick reply.</p>
<p>The issue I created was that I had changes in BOTH local and origin repos.  The local changes were significantly more extensive (new files and other edits to almost all files).  The origin edit was impulsive one: updating README file.</p>
<p>The task I was trying to achieve:<br>
update local README from origin/master AND keep all local changes.</p>
<p>When I did</p>
<pre><code class="lang-auto">git fetch &amp;&amp; git merge origin/master
</code></pre>
<p>it showed deletions to all my local files (disastrous) and addition to README (desirable).</p>
<p>Question remains: can I update a single file without touching anything else?</p>
<p>Thanks you.<br>
Radovan</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Ah, I think there might be a misunderstanding about <code>git diff</code> here. A diff does not show you what happens if you would merge, it just shows you the differences between the branches as they currently are.</p>
<p>A merge would try to bring the changes on both branches <em>since the last common ancestor commit</em> together: Both your new local files, and the README change in the GitHub repository. So yes, it sounds like a <code>git merge origin/master</code> would have gotten you what you wanted. :slight_smile:</p>
<p>A problem that can happen when merging is a <em>merge conflict</em>, e.g. if the same section of a file has been changed on both branches, or a file has been changed on one branch and deleted on the other. In that case git will notify you about the conflict and you’ll have to resolve it manually, deciding what you want to keep.</p>
<p>I recommend reading the <a href="https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging" rel="nofollow noopener">Basic Branching and Merging</a> chapter of the Git Book to get a better idea of what merging does.</p>
